glyphs2fontir/src/plist.rs

```diff
@@ -0,0 +1,450 @@
+                                        if (b'0'..=b'7').contains(&b1) {
+                                            let oct =
+                                                (b - b'0') * 64 + (b1 - b'0') * 8 + (b2 - b'0');
+                                            buf.push(oct as char);
+                                            ix += 2;
+                                            cow_start = ix + 1;
+                                        } else {
+                                            return Err(Error::UnknownEscape);
+                                        }
+                                    } else {
+                                        return Err(Error::UnknownEscape);
+                                    }
+                                }
+                            }
+                            ix += 1;
+                        }
+                        _ => ix += 1,
+                    }
+                }
+                Err(Error::UnclosedString)
+            }
+            _ => {
+                if is_alnum(b) {
+                    let mut ix = start + 1;
+                    while ix < s.len() {
+                        if !is_alnum(s.as_bytes()[ix]) {
+                            break;
+                        }
+                        ix += 1;
+                    }
+                    Ok((Token::Atom(&s[start..ix]), ix))
+                } else {
+                    Err(Error::UnexpectedChar(s[start..].chars().next().unwrap()))
+                }
+            }
+        }
+    }
+
+    fn try_into_string(self) -> Result<String, Error> {
+        match self {
+            Token::Atom(s) => Ok(s.into()),
+            Token::String(s) => Ok(s.into()),
+            _ => Err(Error::NotAString),
+        }
+    }
+
+    fn expect(s: &str, ix: usize, delim: u8) -> Option<usize> {
+        let ix = skip_ws(s, ix);
+        if ix < s.len() {
+            let b = s.as_bytes()[ix];
+            if b == delim {
+                return Some(ix + 1);
+            }
+        }
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{fs, path::Path};
+
+    use super::Plist;
+
+    fn wghtvar_glyphs_file() -> String {
+        fs::read_to_string(Path::new("../resources/testdata/WghtVar.glyphs")).unwrap()
+    }
+
+    #[test]
+    fn tracks_slices() {
+        let raw_plist = wghtvar_glyphs_file();
+        let Plist::Dictionary(plist, range) = Plist::parse(&raw_plist).unwrap() else {
+            panic!("Root of .glyphs should be a dict");
+        };
+        assert_eq!(0usize..raw_plist.trim_end().len(), range);
+
+        let Plist::Integer(upem, range) = plist.get("unitsPerEm").unwrap() else {
+            panic!("upem isn't an int?!");
+        };
+        assert_eq!(*upem, 1000i64);
+        assert_eq!("1000", raw_plist[range.start..range.end].trim());
+
+        let Plist::Array(glyphs, _) = plist.get("glyphs").unwrap() else {
+            panic!("upem isn't an int?!");
```

Comment:
copypasta from the previous panic! statement?
should read something like 

```suggestion
            panic!("'glyphs' isn't an array?!");
```

Reply:
Oh yes ty!